c_emulator/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+set(EMULATOR_COMMON_SRCS
+    riscv_config.h
+    riscv_platform.c
+    riscv_platform.h
+    riscv_platform_impl.c
+    riscv_platform_impl.h
+    riscv_prelude.c
+    riscv_prelude.h
+    riscv_sail.h
+    riscv_sim.c
+    riscv_softfloat.c
+    riscv_softfloat.h
+)
+
+foreach (arch IN ITEMS "rv32" "rv32d" "rv64" "rv64f" "rv32_rvfi" "rv32d_rvfi" "rv64_rvfi" "rv64f_rvfi")
```

Comment:
Should we use variants here as well to make it easier to add the other targets in the future?